### PR TITLE
feat /added weather component and weather route started

### PR DESCRIPTION
**Pull request by jrob112**

### CODE REVIEW

File: client/components/Weather.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import axios from 'axios';
+import { Typography, Box } from '@mui/material';
+
+const Weather = () => {
+  const [weatherData, setWeatherData] = useState(null);
+  const weatherRef = useRef(weatherData);
+
+  // Get api/weather
+  const getWeatherData = () => {
+    axios.get('api/weather')
```
Comment (AlexPHebert2000) [inline client/components/Weather.jsx]: This should be '/weather'. there is no /api endpoint

File: client/components/Weather.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  };
+
+
+  useEffect(() => {
+    getWeatherData();
+  }, [weatherRef]);
+
+  // conversion to Fahrenheit
+  const toFahrenheit = (celsius) => {
+    return (celsius * 9/5) + 32;
```
Comment (AlexPHebert2000) [inline client/components/Weather.jsx]: Use Math.floor(((celsius * 9/5) + 32) * 100) / 100 to round to the nearest hundredth of a degree

File: client/components/Weather.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  };
+
+  return (
+    <Box sx={styleOrangeBox}>
```
Comment (AlexPHebert2000) [inline client/components/Weather.jsx]: We dont have a 'styleOrangeBox' for this project

File: server/routes/weather.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+const express = require('express');
+const axios = require('axios');
+const weather = express.Router();
+const { GOOGLE_MAPS_API_KEY, VISUAL_CROSSING_API_KEY } = process.env;
+
+weather.get('/weather', (req, res) => {
+  const userIpAddress = req.clientIp;
```
Comment (AlexPHebert2000) [inline server/routes/weather.js]: request IP middleware is not installed on this project. We used it in the previous project to get the user's location. The weather should be based on the location of the hike so the ip address should not be necessary in this project

**Review by AlexPHebert2000 — CHANGES_REQUESTED** [review]
Please take a look at the comments I left and resolve them

**Review by AlexPHebert2000 — APPROVED** [review]

**Review by uhrinuh — APPROVED** [review]

**Pull request merged**